### PR TITLE
feat(ConfigReader): create data structure from all json

### PR DESCRIPTION
**Pull request by atqaamir**

### CODE REVIEW

File: common/config_reader.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import json
+
+
+class ConfigReader:
+    def __init__(self, filename):
+        self.data = self.create_data(filename)
```
Comment (Nabeel07925) [inline common/config_reader.py]: Static method should be called by ClassName instead of self

File: common/config_reader.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    @staticmethod
+    def create_data(filename):
+        """
+
+        :param filename: name of file containing simulation data (base-config)
+        :return: concatenated data structure of all json files
+        :purpose: it iterates over file names in base-config and concatenates all data into one data structure
```
Comment (Nabeel07925) [inline common/config_reader.py]: Purpose should be mentioned in the first line of docstring without writing :purpose

File: common/config_reader.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+        with open("../data/configs/app_config/"+filename) as f:
+            data = json.load(f)
+
+        for i in range(len(data["files"])):
+
+            with open("../data/" + list(data["files"].keys())[i] + "/" + list(data["files"].values())[i]) as f:
+                data["files"][list(data["files"].keys())[i]] = json.load(f)
+
+        return data
```
Comment (Nabeel07925) [inline common/config_reader.py:52]: This data needs to be stored in the config reader class as a static variable.

File: common/config_reader.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        else:
+            raise Exception("Object already exists")
+
+    @staticmethod
+    def get_data(filename="base-configuration.json"):
+        if ConfigReader.__data == "":
+            ConfigReader(filename)
+        return ConfigReader.__data
+
+    def create_data(self, filename):
```
Comment (internaccounts123) [inline common/config_reader.py]: make this function private since it would only be called from inside the class.

File: common/config_reader.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        Virtually private constructor
+        :param filename: json (base-config) containing basic config of simulation and all other json file names 
+        """
+        if ConfigReader.__data == "":
+            ConfigReader.__data = self.create_data(filename)
+        else:
+            raise Exception("Object already exists")
+
+    @staticmethod
+    def get_data(filename="base-configuration.json"):
```
Comment (internaccounts123) [inline common/config_reader.py]: argument would be a key instead of a file.

**Review by internaccounts123 — APPROVED** [review]
[OK]

**Pull request merged**